### ISSUE TITLE
Bug Report: TCK endTest() MqttException fails to assign host = null in hostOffline()

### ISSUE THREAD

**Issue by rpiper** [eclipse-sparkplug/sparkplug#414]
After running through the TCK 3.0.0 Edge Node tests several times, I noticed that after completing each test, the next test would fail until I click Reset Test and started it again. After examining the log, I can see that an exception is being thrown in `utilities.getHostApps().hostOffline()` when called from a given test's `void endTest(Map<String, String> results)` handler.

The log shown an MqttException in endTest:
```
2022-12-16 13:10:30,586 - Test end requested for Edge SessionEstablishment 
2022-12-16 13:10:31,597 - endTest
org.eclipse.paho.client.mqttv3.MqttException: Timed out waiting for a response from the server
	at org.eclipse.paho.client.mqttv3.internal.Token.waitForCompletion(Token.java:112)
	at org.eclipse.paho.client.mqttv3.MqttToken.waitForCompletion(MqttToken.java:67)
	at org.eclipse.sparkplug.tck.utility.HostApplication.send(HostApplication.java:144)
	at org.eclipse.sparkplug.tck.utility.HostApplication.hostOffline(HostApplication.java:213)
	at org.eclipse.sparkplug.tck.test.edge.SessionEstablishmentTest.endTest(SessionEstablishmentTest.java:206)
	at org.eclipse.sparkplug.tck.test.TCK.endTest(TCK.java:160)
	at org.eclipse.sparkplug.tck.test.TCK.endTest(TCK.java:148)
	at org.eclipse.sparkplug.tck.test.edge.SessionEstablishmentTest.checkDBirth(SessionEstablishmentTest.java:986)
	at org.eclipse.sparkplug.tck.test.edge.SessionEstablishmentTest.publish(SessionEstablishmentTest.java:365)
	at org.eclipse.sparkplug.tck.test.TCK.publish(TCK.java:227)
	at org.eclipse.sparkplug.tck.PublishInterceptor.onInboundPublish(PublishInterceptor.java:89)
	at com.hivemq.extensions.handler.IncomingPublishHandler$PublishInboundInterceptorTask.apply(IncomingPublishHandler.java:304)
	at com.hivemq.extensions.handler.IncomingPublishHandler$PublishInboundInterceptorTask.apply(IncomingPublishHandler.java:282)
	at com.hivemq.extensions.executor.task.PluginTaskExecutor$PluginTaskExecutorRunnable.runInOutTask(PluginTaskExecutor.java:332)
	at com.hivemq.extensions.executor.task.PluginTaskExecutor$PluginTaskExecutorRunnable.runTask(PluginTaskExecutor.java:302)
	at com.hivemq.extensions.executor.task.PluginTaskExecutor$PluginTaskExecutorRunnable.executeTask(PluginTaskExecutor.java:251)
	at com.hivemq.extensions.executor.task.PluginTaskExecutor$PluginTaskExecutorRunnable.run(PluginTaskExecutor.java:164)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
2022-12-16 13:10:31,598 - Summary Test Results for Edge SessionEstablishment 
```

The result is that the underlying client does publish the host's death certificate, but then hits this timeout exception and never marks the host as null in the `hostOffline()` method in HostApplication.java:
```
public void hostOffline() throws MqttException {
    if (host == null) {
        logger.info("no host application");
        return;
    }
    // send OFFLINE state message
    send(deathPayload);
    logger.info("Host " + hostApplicationId + " successfully stopped");
    host.disconnect();
    host.close();
    host = null;
}
```

Because the host never gets marked as null at the end of the test, when the next test's constructor calls `utilities.getHostApps().hostOnline(hostApplicationId, true)` the host doesn't get published as online because `(host != null)`:

```
public void hostOnline(String host_application_id, boolean forceReset) throws MqttException {
    if (host != null) {
        logger.info("host application in use");
        return;
    }
    hostPrepare(host_application_id, forceReset);
    hostSendOnline();
    logger.info("Host " + host_application_id + " successfully online");
}
```

I see this on every call to `hostOffline()` made within the test's `void endTest(Map<String, String> results)` handler. All other publishes seem to work fine. Is it maybe due to calling `theTCK.endTest()` from within the `void publish(String clientId, PublishPacket packet)` callback from the HiveMQ monitor?

**Comment by rpiper** [comment]
I also confirmed that `(host != null)` when `utilities.getHostApps().hostOnline(hostApplicationId, true)` is called at the start of the next test from seeing this in the log.
```
2022-12-16 13:10:47,263 - Is Host online? false
2022-12-16 13:10:47,263 - Creating host application
2022-12-16 13:10:47,263 - host application in use
```

**Comment by rpiper** [comment]
I've replaced the sparkplug-tck-3.0.0.zip I was building myself from develop with the official version just released with Sparkplug TCK 3.0.0, and this problem seems to have gone away. Maybe it was a permissions or signing issue.
